Error with latest Xcode: Directory.cpp: '~Iterator' has a non-throwing exception

A semi-recent change to Xcode and the MacOS Command Line Tools cause the nupic.core build to fail.

Currently on my machine with Xcode 10.2.1:

```
/Users/mlewis/nta/nupic.core/src/nupic/os/Directory.cpp:258:3: error: '~Iterator' has a non-throwing exception specification but can still throw [-Werror,-Wexceptions]
  NTA_CHECK(res == 0) << "Couldn't close directory."
```

When I revert to an older command line tools from Fall 2018, the build works.

We should change this destructor to be non-throwing.

--- a/src/nupic/os/Directory.cpp
+++ b/src/nupic/os/Directory.cpp
@@ -255,8 +255,10 @@ void Iterator::init(const std::string &path) {
 Iterator::~Iterator() {
   apr_status_t res = ::apr_dir_close(handle_);
   ::apr_pool_destroy(pool_);
-  NTA_CHECK(res == 0) << "Couldn't close directory."
-                      << " OS num: " << APR_TO_OS_ERROR(res);
+  if (res != 0) {
+    NTA_WARN << "Couldn't close directory."
+             << " OS num: " << APR_TO_OS_ERROR(res);
+  }
 }
 
 void Iterator::reset() {

